src/cmd/run.js

```diff
@@ -42,16 +44,28 @@ export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir,
     onSourceChange = defaultSourceWatcher,
+    desktopNotifications = notifier,
   }: WatcherCreatorParams
  ): Watchpack {
   return onSourceChange({
     sourceDir,
     artifactsDir,
     onChange: () => {
       log.debug(`Reloading add-on ID ${addonId}`);
+
       return client.reloadAddon(addonId)
+        .then(() => {
+          desktopNotifications({
```

Comment:
@saintsebastian :+1: yeah, I think that I agree with @kumar303 on this, by default we should only show a desktop notification when there is an error. 

(I would be ok to enable the 'desktop notification on success' with an additional cli option, but we can also keep this PR simple and evaluate to introduce the additional cli option and evaluate its correct behavior in a follow up)